### PR TITLE
Refactoring and many other changes

### PR DESCRIPTION
**Pull request by msp1974**
@maciej-or I have taken a look at how this integration is structured and some of the issues raised and done some retructuring which I think (hope!) will both resolve a number of the issues but also make simpler to manage/maintain/enhance going forward.

As this is your great work, please don't think I am trying to take over but want to contribute to a great integration that I use and not create other forks.

I have created a document in this PR called CHANGES.md to explain what I have done and also been running this in the following scenarios without issue:

1. DVR with 4 analog cameras
2. DVR with 4 analog cameras and 1 IP camera
3. DVR with 4 analog cameras as 1 instance and a single IP camera as another instance.

I hope you can acceprt this PR and the code changes are understandable but please reach out if there are any bits to discuss.  In the CHANGES.md document, there are some TODOs which I will also work on before merging this but wanted to see if you were happy with this PR first and then it can be finalised.

Mark

### CODE REVIEW

**Comment by msp1974** [comment]
OK @maciej-or , this should now be ready to test.  The version number will need updating before release if you are happy with it.  I have also updated the changes document to reflect that the TODOs/Known issues are now resolved and total changes made.

File: custom_components/hikvision_next/isapi.py [hunk 1]
```diff
@@ -40,309 +41,518 @@
...
+            url = f"System/Video/inputs/channels/{channel_id}/{slug}"
+        else:
+            # ISAPI/Smart/{event}/{channel_id}
+            url = f"Smart/{slug}/{channel_id}"
+        return url
 
-    async def _get_channel_events_capabilities(self, channel_id: str) -> list[str]:
-        """Get available NVR channel events."""
+    async def get_camera_streams(self, channel_id: int) -> list[CameraStreamInfo]:
+        # TODO - Improve efficiency of this by reading once and adding to each camera
```
Comment (maciej-or) [inline custom_components/hikvision_next/isapi.py:462]: Do you have ISAPI documentation? If not I could share. Maybe there we will find better way?

File: custom_components/hikvision_next/notifications.py [hunk 2]
```diff
@@ -52,60 +60,127 @@ async def post(self, request: web.Request):
...
+                if headers.get(CONTENT_TYPE) in CONTENT_TYPE_XML:
+                    xml = part.text
+                if headers.get(CONTENT_TYPE) == CONTENT_TYPE_IMAGE:
+                    _LOGGER.debug("image found")
+
+        if not xml:
+            raise ValueError(f"Unexpected event Content-Type {content_type_header}")
+        return xml
+
+    def parse_event_notification(self, xml: str) -> AlertInfo:
```
Comment (maciej-or) [inline custom_components/hikvision_next/notifications.py:92]: because parse_event_notification contains keys related to ISAPI  better 
keep it in isapi.py

**Review by maciej-or — COMMENTED** [review]
Thanks for great contribution. I really like the refactoring and diagnostic data. I tested integration on my setup. NVR works perfectly, standalone ip camera fails with one request. I will investigate that soon and let you know. I'm going to install this branch on my main HA and test for a couple of days.
The PR generally looks good to me, but please follow HA style guidelines https://developers.home-assistant.io/docs/development_guidelines
I have installed development environment according to https://developers.home-assistant.io/docs/development_environment and IDE shows all rules violations.

**Comment by msp1974** [comment]
OK, I have fixed linting errors (I was using a non dev-container setup which doesn't show these) and the other items listed.  I have also removed the HA event firing for now as need to think more through how to do with multiple instances of the integration configured.  Will likely work on this for another PR if thats ok.

Did you find the error with the single IP camera?

**Comment by maciej-or** [comment]
Cool, I will manage my single camera issue and we could release. The PR becomes massive so yea create a new branch for next updates.

**Pull request merged**